src/libs/SidebarUtils.js

```diff
@@ -16,11 +16,20 @@ import * as CollectionUtils from './CollectionUtils';
 // Session also can remain stale because the only way for the current user to change is to sign out and sign in, which would clear out all the Onyx
 // data anyway and cause SidebarLinks to rerender.
 
-let reports;
+const chatReports = {};
+const iouReports = {};
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
-    waitForCollectionCallback: true,
-    callback: val => reports = val,
+    callback: (report, key) => {
+        if (!report) {
+            delete iouReports[key];
+            delete chatReports[key];
+        } else if (ReportUtils.isIOUReport(report)) {
+            iouReports[key] = report;
+        } else {
+            chatReports[key] = report;
+        }
+    },
```

Comment:
NAB: What do you think about having some helper function that you can use to make the connection to avoid repeating this code?

```js
const chatReports = {};
const iouReports = {};
connectReports(chatReports, iouReports);
```

```js
function connectReports(chatReports, iouReports) {
    Onyx.connect({
        key: ONYXKEYS.COLLECTION.REPORT,
        callback: (report, key) => {
            if (!report) {
                delete iouReports[key];
                delete chatReports[key];
            } else if (ReportUtils.isIOUReport(report)) {
                iouReports[key] = report;
            } else {
                chatReports[key] = report;
            }
        },
    });
}
```

then use it in `src/libs/actions/IOU.js` too... not sure if it is worth the effort though.

Reply:
Hmmm. I think we can leave it as-is for now. We don't have helper functions for other connections (although maybe we should)